io/net/infiniband/dapl.py

```diff
@@ -0,0 +1,163 @@
+                                     password=self.peer_password)
+        self.peer_interface = self.remotehost.get_interface_by_ipaddr(self.peer_ip).name
+        self.peer_networkinterface = NetworkInterface(self.peer_interface,
+                                                      self.remotehost)
+
+        if detected_distro.name == "Ubuntu":
+            cmd = "service ufw stop"
+        # FIXME: "redhat" as the distro name for RHEL is deprecated
+        # on Avocado versions >= 50.0.  This is a temporary compatibility
+        # enabler for older runners, but should be removed soon
+        elif detected_distro.name in ['rhel', 'fedora', 'redhat']:
+            cmd = "systemctl stop firewalld"
+        elif detected_distro.name == "SuSE":
+            if detected_distro.version == 15:
+                cmd = "systemctl stop firewalld"
+            else:
+                cmd = "rcSuSEfirewall2 stop"
+        elif detected_distro.name == "centos":
+            cmd = "service iptables stop"
+        else:
+            self.cancel("Distro not supported")
+        if process.system(cmd, ignore_status=True, shell=True) != 0:
+            self.cancel("Unable to disable firewall")
+        output = self.session.cmd(cmd)
+        if not output.exit_status == 0:
+            self.cancel("Unable to disable firewall on peer")
+
+    def test(self):
+        """
+        Test dapl
+        """
+        if self.peer_networkinterface.set_mtu(self.mtu) is not None:
+            self.fail("Failed to set mtu in peer")
+        if self.networkinterface.set_mtu(self.mtu) is not None:
+            self.fail("Failed to set mtu in host")
+        self.log.info(self.test_name)
+        logs = "> /tmp/ib_log 2>&1 &"
+        cmd = " timeout %s dapltest %s %s" % (self.timeout, self.pprm, logs)
+        output = self.session.cmd(cmd)
+        if not output.exit_status == 0:
+            self.fail("SSH connection (or) Server command failed")
+        time.sleep(5)
+        if self.hprm != "None":
+            self.log.info("Client data - %s(%s)" % (self.test_name, self.hprm))
+            cmd = "timeout %s dapltest %s" % (self.timeout, self.hprm)
+            if process.system(cmd, shell=True, ignore_status=True) != 0:
+                self.fail("Client command failed")
+            time.sleep(5)
+        self.log.info("Server data - %s(%s)" % (self.test_name, self.pprm))
+        cmd = "timeout %s cat /tmp/ib_log && rm -rf /tmp/ib_log" \
+            % (self.timeout)
+        output = self.session.cmd(cmd)
+        if not output.exit_status == 0:
+            self.fail("Server output retrieval failed")
+
+    def tearDown(self):
+        """
+        unset ip
+        """
+        if self.networkinterface.set_mtu('1500') is not None:
+            self.fail("Failed to set mtu in host")
+        if self.peer_networkinterface.set_mtu('1500') is not None:
+            self.fail("Failed to set mtu in peer")
+        self.remotehost.remote_session.quit()
```

Comment:
@manvanthar quit the self.session after using connect() . and add restore_from_backup() if you want restore the config file.
and here not added  ip uunset things.
some time we unable to ping peer machine in the end if some error came to set_mtu "1500" , so to avoid that you can add it will set_mtu through public ip of peer if its fail.

Reply:
addressed in new patch